### PR TITLE
Add tests for validate BLS Key endpoint - Closes #1288

### LINKED ISSUE #1288: Add tests for validate BLS Key endpoint
### Description

Lisk Service adds support to let end users test their BLS Keys with https://github.com/LiskHQ/lisk-service/issues/1287. The implementation needs to be tested on CI to ensure code quality.

### Acceptance Criteria

- Unit tests are implemented
- Integration tests are implemented

#### Endpoints
**HTTP**:  `POST /validator/validateBLSKey`
**RPC**:    `post.validator.validateBLSKey`

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1288

### How was it solved?
- [x] Unit tests are implemented. Not required as it is simply forwarding the call.
- [x] Integration tests are implemented
### How was it tested?
- [x] Local

### CODE REVIEW

File: tests/integration/api_v3/http/postTransactions.test.js [hunk 1]
```diff
@@ -26,8 +26,8 @@ const {
 } = require('../../../schemas/api_v3/transaction.schema');
 
 const baseUrl = config.SERVICE_ENDPOINT;
-const baseUrlV2 = `${baseUrl}/api/v3`;
-const endpoint = `${baseUrlV2}/transactions`;
+const baseUrlV3 = `${baseUrl}/api/v3`;
+const endpoint = `${baseUrlV3}/transactions`;
 
 // TODO: Enable test cases once given issue is resolved https://github.com/LiskHQ/lisk-sdk/issues/7172
 xdescribe('Post transactions API', () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/postTransactions.test.js:33]: Please remove the TODO and enable the tests.

File: tests/integration/api_v3/http/validateBLSKey.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const { validateBLSKeySchema } = require('../../../schemas/api_v3/validatorSchema.schema');
+
+const baseUrl = config.SERVICE_ENDPOINT;
+const baseUrlV3 = `${baseUrl}/api/v3`;
+const endpoint = `${baseUrlV3}/validator/validateBLSKey`;
+
+describe('validate BLS Key API', () => {
+	it('Returns true for valid blsKey and proofOfPossession pair', async () => {
+		const blsKey = 'b301803f8b5ac4a1133581fc676dfedc60d891dd5fa99028805e5ea5b08d3491af75d0707adab3b70c6a6a580217bf81';
+		const proofOfPossession = '88bb31b27eae23038e14f9d9d1b628a39f5881b5278c3c6f0249f81ba0deb1f68aa5f8847854d6554051aa810fdf1cdb02df4af7a5647b1aa4afb60ec6d446ee17af24a8a50876ffdaf9bf475038ec5f8ebeda1c1c6a3220293e23b13a9a5d26';
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/validateBLSKey.test.js:28]: Please add these values as constants and then reuse them in both the test files.

File: tests/integration/api_v3/http/validateBLSKey.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		const response = await api.post(
+			`${endpoint}?blsKey=${blsKey}&proofOfPossession=${proofOfPossession}`,
+		);
+
+		expect(response.data).toMap(validateBLSKeySchema);
+		expect(response.data.isValid).toEqual(false);
+	});
+
+	it('Returns true for wrong proofOfPossession message', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/validateBLSKey.test.js:48]: Also, please add an additional test case to verify that the application returns an error when an unexpected param is supplied in the request. Kindly do this for both the HTTP/RPC tests.
```suggestion
	it('Returns false for wrong proofOfPossession message', async () => {
```

File: tests/integration/api_v3/http/validateBLSKey.test.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+describe('validate BLS Key API', () => {
+	it('Returns true for valid blsKey and proofOfPossession pair', async () => {
+		const blsKey = 'b301803f8b5ac4a1133581fc676dfedc60d891dd5fa99028805e5ea5b08d3491af75d0707adab3b70c6a6a580217bf81';
+		const proofOfPossession = '88bb31b27eae23038e14f9d9d1b628a39f5881b5278c3c6f0249f81ba0deb1f68aa5f8847854d6554051aa810fdf1cdb02df4af7a5647b1aa4afb60ec6d446ee17af24a8a50876ffdaf9bf475038ec5f8ebeda1c1c6a3220293e23b13a9a5d26';
+		const response = await api.post(
+			`${endpoint}?blsKey=${blsKey}&proofOfPossession=${proofOfPossession}`,
+		);
+
+		expect(response.data).toMap(validateBLSKeySchema);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/validateBLSKey.test.js:33]: Please also validate that `response` also adheres to the standard response envelope.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Kindly apply the following recommendations to both the test files.

File: tests/integration/api_v3/http/validateBLSKey.test.js (hunk 2 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/validateBLSKey.test.js:28]: applied.

File: tests/integration/api_v3/http/validateBLSKey.test.js (hunk 3 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/validateBLSKey.test.js:48]: applied.

File: tests/integration/api_v3/http/postTransactions.test.js (hunk 1 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/postTransactions.test.js]: WIP

File: tests/integration/api_v3/http/validateBLSKey.test.js (hunk 4 above)
Comment (sameersubudhi) [reply tests/integration/api_v3/http/validateBLSKey.test.js:33]: Kindly create a response envelope for endpoints where the schema varies. Please refer the implementation for an endpoint such as `/network/status` or `/fees`.

File: tests/integration/api_v3/http/postTransactions.test.js [hunk 5]
```diff
@@ -26,28 +26,27 @@ const {
...
 	it('Post transaction succesfully', async () => {
 		const postTransaction = await api.post(
-			`${endpoint}?transaction='0802100018002080c2d72f2a200fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a322e0880a0cebdf3ef171214aebd99f07218109162a905d0e0c91e58bedc83c51a0e746f6b656e207472616e736665723a40a30998db4e96911a3d784b0a01b817baf64ec9745d7c67bc537b060eae27ec7ab85ffbb36d76f2b62eaf796deba0407255967506cb01764220e2e6ce66183d07'`,
+			`${endpoint}?transaction=0a05746f6b656e12087472616e7366657218002080c2d72f2a20a3f96c50d0446220ef2f98240898515cbba8155730679ca35326d98dcfb680f032270a0800000000000000001080c2d72f1a1474e3ba5ade3e94451bd4de9d19917c8e6eff624d22003a405fd81dc9d48c04f3a197ea9d57236ff208388a57f7686f19f7917e92679d4da162b97b357f941e3a383f6a795fa9710da5c9dc5eda7007e5b4bbe6551e59360c`,
 		);
 		expect(postTransaction).toMap(postTransactionSchema);
 	});
 
-	it('throws error when posting invalid binary transaction', async () => {
+	it.only('throws error when posting invalid binary transaction', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/postTransactions.test.js:40]: Please do not commit exclusive tests.
```suggestion
	it('throws error when posting invalid binary transaction', async () => {
```

File: tests/integration/api_v3/rpc/postTransactions.test.js [hunk 6]
```diff
@@ -32,18 +32,20 @@ const {
...
 		);
-		expect(response).toMap(postTransactionSchema);
+		const { result } = response;
+
+		expect(result).toBeInstanceOf(Object);
+		expect(result).toMap(postTransactionSchema);
 	});
 
-	it('invalid binary transaction -> empty response', async () => {
+	it.only('invalid binary transaction -> empty response', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/postTransactions.test.js:48]: Same as above.
```suggestion
	it('invalid binary transaction -> empty response', async () => {
```

File: tests/integration/api_v3/http/postTransactions.test.js (hunk 5 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/postTransactions.test.js:40]: WIP commit, kept a pointer for myself to fix.

File: tests/integration/api_v3/rpc/postTransactions.test.js (hunk 6 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/postTransactions.test.js:48]: WIP commit, kept a pointer for myself to fix.

File: tests/constants.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2019 Lisk Foundation
```
Comment (nagdahimanshu) [inline tests/constants.js:3]:
```suggestion
 * Copyright © 2022 Lisk Foundation
```

Comment (priojeetpriyom) [reply tests/constants.js:3]: There are many files with 2019. Will refactor all of those at once after applying other comments.

Comment (priojeetpriyom) [reply tests/constants.js]: As explained by Sameer, updated this one only.

File: tests/integration/api_v3/http/postTransactions.test.js (hunk 1 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/postTransactions.test.js]: applied.

File: tests/integration/api_v3/rpc/postTransactions.test.js (hunk 6 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/postTransactions.test.js]: removed.

File: tests/integration/api_v3/http/postTransactions.test.js (hunk 5 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/postTransactions.test.js]: removed.

File: tests/integration/api_v3/http/validateBLSKey.test.js (hunk 3 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/validateBLSKey.test.js]: applied.

File: services/blockchain-indexer/shared/dataService/business/validator/index.js [hunk 8]
```diff
@@ -14,7 +14,7 @@
  *
  */
 const { getValidator } = require('./validatorInfo');
-const { validateBLSKey } = require('./validateBLSKey');
+const { validateBLSKey } = require('./validatorValidateBLSKey');
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/dataService/business/validator/index.js:17]: Please do not rename the file to `validatorValidateBLSKey` since it is already within the `validator` directory. The renaming was supposed to be only for the test files. Please revert this change.
```suggestion
const { validateBLSKey } = require('./validateBLSKey');
```

File: tests/integration/api_v3/rpc/postTransactions.test.js [hunk 9]
```diff
@@ -32,26 +33,26 @@ const {
...
 			},
 		);
-		expect(response).toMap(postTransactionSchema);
+		const { result } = response;
+
+		expect(result).toBeInstanceOf(Object);
+		expect(result).toMap(postTransactionSchema);
 	});
 
 	it('invalid binary transaction -> empty response', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/postTransactions.test.js:49]:
```suggestion
	it('Invalid binary transaction -> empty response', async () => {
```

File: tests/integration/api_v3/rpc/postTransactions.test.js [hunk 10]
```diff
@@ -32,26 +33,26 @@ const {
...
-		const response = await postTransaction({ transaction: 'a200fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a32250880c2d72f1214b49074a2eb04e7611908985f02c12fb7cd488d451a0874657374207478733a4065faed7b49d1ee63730cbb545ea25e5036' }).catch(e => e);
-		expect(response).toMap(emptyResponseSchema);
-		const { result } = response;
-		expect(result).toMap(emptyResultEnvelopeSchema);
+		const response = await postTransaction({ transaction: INVALID_TRANSACTION }).catch(e => e);
+		expect(response).toMap(jsonRpcEnvelopeSchema);
 	});
 
 	it('invalid query parameter -> -32602', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/postTransactions.test.js:54]:
```suggestion
	it('Invalid query parameter -> -32602', async () => {
```

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+		expect(result.data).toMap(validateBLSKeySchema);
+		expect(result.data.isValid).toEqual(false);
+	});
+
+	it('No param -> invalid param', async () => {
+		const response = await validateBLSKey({});
+		expect(response).toMap(invalidParamsSchema);
+	});
+
+	it('extra param -> invalid param', async () => {
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:72]:
```suggestion
	it('Invalid param -> invalid param', async () => {
```

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+		const response = await validateBLSKey({
+			blsKey: BLS_KEY.valid,
+			proofOfPossession: PROOF_OF_POSSESSION.valid,
+			extra_param: EXTRA_PARAM,
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:76]:
```suggestion
			invalidParam: 'value',
```

File: services/gateway/apis/http-version3/methods/validatorValidateBLSKey.js [hunk 13]
```diff
@@ -13,7 +13,7 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-const validateBLSKeySources = require('../../../sources/version3/validateBLSKey');
+const validateBLSKeySources = require('../../../sources/version3/validatorValidateBLSKey');
```
[the remark this answers is not in the document]
Comment (sameersubudhi) [reply services/gateway/apis/http-version3/methods/validatorValidateBLSKey.js:16]: You can keep the changes within the Gateway microservice.

File: services/blockchain-indexer/shared/dataService/business/validator/index.js (hunk 8 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/validator/index.js:17]: applied.

File: tests/integration/api_v3/http/postTransactions.test.js [hunk 14]
```diff
@@ -26,28 +27,32 @@ const {
...
+const endpoint = `${baseUrlV3}/transactions`;
 
-// TODO: Enable test cases once given issue is resolved https://github.com/LiskHQ/lisk-sdk/issues/7172
-xdescribe('Post transactions API', () => {
+describe('Post transactions API', () => {
 	it('Post transaction succesfully', async () => {
 		const postTransaction = await api.post(
-			`${endpoint}?transaction='0802100018002080c2d72f2a200fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a322e0880a0cebdf3ef171214aebd99f07218109162a905d0e0c91e58bedc83c51a0e746f6b656e207472616e736665723a40a30998db4e96911a3d784b0a01b817baf64ec9745d7c67bc537b060eae27ec7ab85ffbb36d76f2b62eaf796deba0407255967506cb01764220e2e6ce66183d07'`,
+			endpoint,
+			{ transaction: '0a05746f6b656e12087472616e7366657218002080c2d72f2a20a3f96c50d0446220ef2f98240898515cbba8155730679ca35326d98dcfb680f032270a0800000000000000001080c2d72f1a1474e3ba5ade3e94451bd4de9d19917c8e6eff624d22003a405fd81dc9d48c04f3a197ea9d57236ff208388a57f7686f19f7917e92679d4da162b97b357f941e3a383f6a795fa9710da5c9dc5eda7007e5b4bbe6551e59360c' },
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/postTransactions.test.js:37]: applied.

File: tests/integration/api_v3/constants/validatorValidateBLSKey.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+	valid: 'b301803f8b5ac4a1133581fc676dfedc60d891dd5fa99028805e5ea5b08d3491af75d0707adab3b70c6a6a580217bf81',
+	invalid: '1301803f8b5ac4a1133581fc676dfedc60d891dd5fa99028805e5ea5b08d3491af75d0707adab3b70c6a6a580217bf81',
+};
+
+const PROOF_OF_POSSESSION = {
+	valid: '88bb31b27eae23038e14f9d9d1b628a39f5881b5278c3c6f0249f81ba0deb1f68aa5f8847854d6554051aa810fdf1cdb02df4af7a5647b1aa4afb60ec6d446ee17af24a8a50876ffdaf9bf475038ec5f8ebeda1c1c6a3220293e23b13a9a5d26',
+	invalid: '18bb31b27eae23038e14f9d9d1b628a39f5881b5278c3c6f0249f81ba0deb1f68aa5f8847854d6554051aa810fdf1cdb02df4af7a5647b1aa4afb60ec6d446ee17af24a8a50876ffdaf9bf475038ec5f8ebeda1c1c6a3220293e23b13a9a5d26',
+};
+
+const EXTRA_PARAM = 'extra_value';
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/integration/api_v3/constants/validatorValidateBLSKey.js:26]: applied.

File: tests/integration/api_v3/http/validatorValidateBLSKey.test.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+		const response = await api.post(endpoint, {}, 400);
+		expect(response).toMap(badRequestSchema);
+	});
+
+	it('extra param -> bad request', async () => {
+		const response = await api.post(endpoint,
+			{
+				blsKey: BLS_KEY.valid,
+				proofOfPossession: PROOF_OF_POSSESSION.valid,
+				extra_param: EXTRA_PARAM,
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/integration/api_v3/http/validatorValidateBLSKey.test.js:78]: applied.

File: tests/integration/api_v3/rpc/postTransactions.test.js [hunk 17]
```diff
@@ -32,26 +33,26 @@ const {
...
 const postTransaction = async params => request(wsRpcUrl, 'post.transactions', params);
 
-// TODO: Enable test cases once given issue is resolved https://github.com/LiskHQ/lisk-sdk/issues/7172
-xdescribe('Method post.transactions', () => {
+describe('Method post.transactions', () => {
 	it('Post transaction succesfully', async () => {
 		const response = await postTransaction(
 			{
-				transaction: '0802100018002080c2d72f2a200fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a32250880c2d72f1214b49074a2eb04e7611908985f02c12fb7cd488d451a0874657374207478733a4065faed7b49d1ee63730cbb545ea25e50361581e35412eeffac3b35746afd1176d2ee1e270e8b072b1ccfad2d64f72918063c383003971600d56b168d6e429f05',
+				transaction: '0a05746f6b656e12087472616e7366657218012080c2d72f2a20a3f96c50d0446220ef2f98240898515cbba8155730679ca35326d98dcfb680f032270a0800000000000000001080c2d72f1a1474e3ba5ade3e94451bd4de9d19917c8e6eff624d22003a403881faa16364f5d6a08fb50736ec4809f4d50bba90a3e00c3b10d4f96b85fe50268bf545e95632d582d4ccac750953a09faa5da9d6d49f441baea7f888e5c007',
```
[the remark this answers is not in the document]
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/postTransactions.test.js:40]: applied.

File: tests/integration/api_v3/rpc/postTransactions.test.js (hunk 9 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/postTransactions.test.js:49]: applied.

File: tests/integration/api_v3/rpc/postTransactions.test.js (hunk 10 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/postTransactions.test.js:54]: applied.

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js (hunk 11 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:72]: applied.

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js (hunk 12 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:76]: applied.

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+	it('Returns false for invalid blsKey', async () => {
+		const response = await validateBLSKey({
+			blsKey: BLS_KEY.INVALID,
+			proofOfPossession: PROOF_OF_POSSESSION.VALID,
+		});
+		const { result } = response;
+
+		expect(result.data).toBeInstanceOf(Object);
+		expect(result.data).toMap(validateBLSKeySchema);
+		expect(result.data.isValid).toEqual(false);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:52]:
```suggestion
		const { result } = response;
		expect(result).toMap(validateBLSKeyGoodRequestSchema);
		expect(result.data).toBeInstanceOf(Object);
		expect(result.data).toMap(validateBLSKeySchema);
		expect(result.data.isValid).toEqual(false);
```

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+	it('Returns true for invalid proofOfPossession message', async () => {
+		const response = await validateBLSKey({
+			blsKey: BLS_KEY.VALID,
+			proofOfPossession: PROOF_OF_POSSESSION.INVALID,
+		});
+		const { result } = response;
+
+		expect(result.data).toBeInstanceOf(Object);
+		expect(result.data).toMap(validateBLSKeySchema);
+		expect(result.data.isValid).toEqual(false);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:64]:
```suggestion
		const { result } = response;
		expect(result).toMap(validateBLSKeyGoodRequestSchema);
		expect(result.data).toBeInstanceOf(Object);
		expect(result.data).toMap(validateBLSKeySchema);
		expect(result.data.isValid).toEqual(false);
```

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+		expect(result.data).toMap(validateBLSKeySchema);
+		expect(result.data.isValid).toEqual(true);
+	});
+
+	it('Returns false for invalid blsKey', async () => {
+		const response = await validateBLSKey({
+			blsKey: BLS_KEY.INVALID,
+			proofOfPossession: PROOF_OF_POSSESSION.VALID,
+		});
+		const { result } = response;
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:48]:
```suggestion
		expect(response).toMap(jsonRpcEnvelopeSchema);
```

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+		expect(result.data).toMap(validateBLSKeySchema);
+		expect(result.data.isValid).toEqual(false);
+	});
+
+	it('Returns true for invalid proofOfPossession message', async () => {
+		const response = await validateBLSKey({
+			blsKey: BLS_KEY.VALID,
+			proofOfPossession: PROOF_OF_POSSESSION.INVALID,
+		});
+		const { result } = response;
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:60]:
```suggestion
		expect(response).toMap(jsonRpcEnvelopeSchema);
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Could you please apply the following suggestions?

File: tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js (hunk 21 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/rpc/validatorValidateBLSKey.test.js:61]: applied.

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍
